### PR TITLE
fix(LocalGraphQLClient): handle middleware and responseReducer

### LINKED ISSUE #1196: using LocalGraphQLClient with passing localQueries for test purposes is not calling the middleware functions
### Package

`graphql-hooks`

### Environment

- `graphql-hooks` version: 6.4.0
- `react` version: 17.0.2
- Browser: Chrome

### Description

using LocalGraphQLClient with passing localQueries for test purposes is not calling the middleware functions and causing the middleware not being tested completely

### How to reproduce

1. Create a GraphQL client and pass a middleware dummy function
2. Use LocalGraphQLClient and define some localQueries to mock
3. Debug and observe the middleware function is not getting called

```jsx
```

### PR DESCRIPTION
**Pull request by G100g**
### What does this PR do?

Let  `LocalGraphQLClient` to handle `middleware`and `responseReducer`

### Related issues

resolve #1196 

### Checklist

- [x] I have checked the [contributing document](../blob/master/CONTRIBUTING.md)
- [x] I have added or updated any relevant documentation
- [x] I have added or updated any relevant tests

### CODE REVIEW

File: packages/graphql-hooks/src/LocalGraphQLClient.ts [hunk 1]
```diff
@@ -41,29 +47,38 @@ class LocalGraphQLClient extends GraphQLClient {
...
+    return super
+      .request<ResponseData, TGraphQLError, TVariables>(operation, options)
       .then(result => {
         if (result instanceof LocalGraphQLError) {
           return { error: result }
         }
-        const { data, errors } = collectErrorsFromObject(result)
-        if (errors.length > 0) {
+        const { data, errors } = collectErrors(result)
+        if (errors && errors.length > 0) {
```
Comment (bilalshareef) [inline packages/graphql-hooks/src/LocalGraphQLClient.ts:81]: **nitpick:** This line can be `if (errors?.length > 0) {`

**Review by bilalshareef — APPROVED** [review]
Looks good to me.

File: packages/graphql-hooks/src/LocalGraphQLClient.ts (hunk 1 above)
Comment (G100g) [reply packages/graphql-hooks/src/LocalGraphQLClient.ts:81]: Thanks for the feedback.
I tried to use it before push the code, but I got this error from TS

image

So I preferred to stay with this version :D

Comment (bilalshareef) [reply packages/graphql-hooks/src/LocalGraphQLClient.ts:81]: Oh okay. The current code looks good to me then. 🙂

**Review by simoneb — APPROVED** [review]

**Pull request merged**